spring-integration-core/src/main/java/org/springframework/integration/graph/MessageSourceNode.java

```diff
@@ -33,9 +33,9 @@
  */
 public class MessageSourceNode extends ErrorCapableEndpointNode implements ReceiveCountersAware {
 
-	private Supplier<ReceiveCounters> receiveCounters;
+	private @Nullable Supplier<ReceiveCounters> receiveCounters;
 
-	public MessageSourceNode(int nodeId, String name, MessageSource<?> messageSource, String output, String errors) {
+	public MessageSourceNode(int nodeId, String name, MessageSource<?> messageSource, @Nullable String output, @Nullable String errors) {
```

Comment:
The `SourcePollingChannelAdapter` does not return null for its `outputChannel`.
Please, revise.